Chore/optimize selection code

JimmFly opened the pull request:
Optimized for code readability and maintainability

doodlewind commented:
Could you elaborate what has been reached through the optimization? Thanks!

doodlewind commented on packages/blocks/src/__internal__/utils/selection.ts:
```diff
@@ -22,6 +22,99 @@ const notStrictCharacterReg = /[^\p{Alpha}\p{M}\p{Nd}\p{Pc}\p{Join_C}]/u;
 const notStrictCharacterAndSpaceReg =
   /[^\p{Alpha}\p{M}\p{Nd}\p{Pc}\p{Join_C}\s]/u;
 
+// function fixCurrentRangeToText(
```
We shouldn't checkin large batch of commented code in the codebase (unless it's a WIP PR), it won't be maintained by others and will be staled.

doodlewind commented on packages/blocks/src/__internal__/utils/selection.ts:
```diff
@@ -49,43 +142,16 @@ function fixCurrentRangeToText(
+      const rect = text.getBoundingClientRect();
+      const NewY = isForward ? rect.bottom - 6 : rect.top + 6;
+      newRange = caretRangeFromPoint(x, NewY);
+      isForward
```
What is the `null` used for here? Plain if-else looks fine here. Is this what you want to express?

``` ts
if (isForward && newRange) {
  forwardSelect(newRange, range);
}
else if (!isForward && newRange) {
  backwardSelect(newRange, range);
}
```

JimmFly commented:
> Could you elaborate what has been reached through the optimization? Thanks!

I modified the code where the if...else nested too deeply and extracted some new functionality to make the code easier to read and maintain. Before that, the deepest layer of the code has 7 layers of nesting, and each function has nearly 100 lines of code at the same time, which makes me unable to understand.

doodlewind commented:
Thanks for the refactoring, hope we can have a more robust selection feature!
